packages/salesforcedx-vscode-core/src/commands/forceCreateApex.ts

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (c) 2017, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import {
+  Command,
+  SfdxCommandBuilder
+} from '@salesforce/salesforcedx-utils-vscode/out/src/cli';
+import * as path from 'path';
+import * as vscode from 'vscode';
+import fs = require('fs');
+import { nls } from '../messages';
+import {
+  CancelResponse,
+  ContinueResponse,
+  ParametersGatherer,
+  SfdxCommandlet,
+  SfdxCommandletExecutor,
+  SfdxWorkspaceChecker
+} from './commands';
+
+function getDirs(srcPath: string) {
+  return fs
+    .readdirSync(srcPath)
+    .map(name => path.join(srcPath, name))
+    .filter(source => fs.lstatSync(source).isDirectory());
+}
+
+function flatten(lists: string[][]) {
+  return lists.reduce((a, b) => a.concat(b), []);
+}
+
+function getDirsRecursive(srcPath: string): string[] {
+  return [
+    srcPath,
+    ...flatten(getDirs(srcPath).map(src => getDirsRecursive(src)))
+  ];
+}
+
+class SelectFilePath implements ParametersGatherer<{}> {
```

Comment:
Why is the type {}? It should be the type that you are returning, which is `data: { template: string, ....}